src/rdkafka_partition.c

```diff
@@ -2816,6 +2856,44 @@ rd_kafka_topic_partition_list_copy (const rd_kafka_topic_partition_list_t *src){
         return dst;
 }
 
+
+/**
+ * @brief Compare two partition lists using partition comparator \p cmp.
+ *
+ * @warning This is an O(Na*Nb) operation.
```

Comment:
warning: i'm going to take this as a license to make a similar short-cut in a toppar list fn i need..